orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java

```diff
@@ -134,96 +136,82 @@ private static long toMicros(Timestamp ts) {
-          ((StringColumnStatistics) columnStats).getMinimum());
+      min = ((StringColumnStatistics) columnStats).getMinimum();
     } else if (columnStats instanceof DecimalColumnStatistics) {
       min = Optional
           .ofNullable(((DecimalColumnStatistics) columnStats).getMinimum())
-          .map(minStats -> {
-            BigDecimal minValue = minStats.bigDecimalValue()
-                .setScale(((Types.DecimalType) column.type()).scale());
-            return Conversions.toByteBuffer(column.type(), minValue);
-          })
+          .map(minStats -> minStats.bigDecimalValue()
+              .setScale(((Types.DecimalType) column.type()).scale()))
           .orElse(null);
     } else if (columnStats instanceof DateColumnStatistics) {
       min = Optional.ofNullable(((DateColumnStatistics) columnStats).getMinimum())
-          .map(minStats -> Conversions.toByteBuffer(column.type(),
-              (int) ChronoUnit.DAYS.between(EPOCH_DAY,
+          .map(minStats ->
+              Math.toIntExact(ChronoUnit.DAYS.between(EPOCH_DAY,
                   EPOCH.plus(minStats.getTime(), ChronoUnit.MILLIS).toLocalDate())))
           .orElse(null);
     } else if (columnStats instanceof TimestampColumnStatistics) {
       TimestampColumnStatistics tColStats = (TimestampColumnStatistics) columnStats;
-      Timestamp minValue = ((Types.TimestampType) column.type()).shouldAdjustToUTC() ?
-          tColStats.getMinimum() : tColStats.getMinimumUTC();
+      Timestamp minValue = tColStats.getMinimumUTC();
       min = Optional.ofNullable(minValue)
-          .map(v -> Conversions.toByteBuffer(column.type(), toMicros(v)))
+          .map(OrcMetrics::toMicros)
           .orElse(null);
     } else if (columnStats instanceof BooleanColumnStatistics) {
       BooleanColumnStatistics booleanStats = (BooleanColumnStatistics) columnStats;
-      min = booleanStats.getFalseCount() > 0 ?
-          Conversions.toByteBuffer(column.type(), false) :
-          Conversions.toByteBuffer(column.type(), true);
+      min = booleanStats.getFalseCount() <= 0;
     }
-    return Optional.ofNullable(min);
+    return Optional.ofNullable(Conversions.toByteBuffer(column.type(), min));
   }
 
   private static Optional<ByteBuffer> fromOrcMax(Types.NestedField column,
                                                  ColumnStatistics columnStats) {
-    ByteBuffer max = null;
+    Object max = null;
     if (columnStats instanceof IntegerColumnStatistics) {
-      IntegerColumnStatistics intColStats = (IntegerColumnStatistics) columnStats;
-      max = column.type().typeId() == Type.TypeID.INTEGER ?
-          Conversions.toByteBuffer(column.type(), (int) intColStats.getMaximum()) :
-          Conversions.toByteBuffer(column.type(), intColStats.getMaximum());
+      max = ((IntegerColumnStatistics) columnStats).getMaximum();
+      if (column.type().typeId() == Type.TypeID.INTEGER) {
+        max = Math.toIntExact((long) max);
+      }
```

Comment:
Minor: I generally prefer not mutating a variable and storing it back so that it is clear that there are two code paths. The ternary operator seems like a cleaner approach.

Reply:
I agree, I generally avoid mutations, but in this case the ternary operator does not work since `max` is an `Object` and it could have either an `int` or `long` (boxed) the JVM defaults to always boxing to a `long`, leading to conversion issues when the value is actually an `INTEGER`.